[2021-04-24 11:45:27,624] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-24 11:45:28,481] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-24 11:45:28,539] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-24 11:45:28,545] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-24 11:45:28,545] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-24 11:45:29,088] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-08-01T00:00:00+00:00
[2021-04-24 11:45:29,371] {standard_task_runner.py:52} INFO - Started process 219 to run task
[2021-04-24 11:45:29,606] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-08-01T00:00:00+00:00', '--job-id', '89', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmpw39es663', '--error-file', '/tmp/tmp98fqe22f']
[2021-04-24 11:45:29,694] {standard_task_runner.py:77} INFO - Job 89: Subtask Stage_songs
[2021-04-24 11:45:31,169] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [running]> on host 88dcd2a12842
[2021-04-24 11:45:32,851] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernest@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-08-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-08-01T00:00:00+00:00
[2021-04-24 11:45:32,894] {stage_redshift.py:38} INFO - Start implementation of StageToRedshiftOperator
[2021-04-24 11:45:33,161] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 39, in execute
    aws_hook = AwsHook(self.aws_credentials_id)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/contrib/hooks/aws_hook.py", line 41, in __init__
    super().__init__(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/amazon/aws/hooks/base_aws.py", line 360, in __init__
    raise AirflowException('Either client_type or resource_type must be provided.')
airflow.exceptions.AirflowException: Either client_type or resource_type must be provided.
[2021-04-24 11:45:33,245] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200801T000000, start_date=20210424T114527, end_date=20210424T114533
[2021-04-24 11:45:34,271] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-26 09:08:48,537] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-26 09:08:48,942] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-26 09:08:48,952] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 09:08:48,955] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-26 09:08:48,955] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 09:08:49,789] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-08-01T00:00:00+00:00
[2021-04-26 09:08:50,039] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-08-01T00:00:00+00:00', '--job-id', '39', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmpq88a6_qd', '--error-file', '/tmp/tmpjsbbi1a_']
[2021-04-26 09:08:50,044] {standard_task_runner.py:77} INFO - Job 39: Subtask Stage_songs
[2021-04-26 09:08:49,914] {standard_task_runner.py:52} INFO - Started process 120 to run task
[2021-04-26 09:08:51,247] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [running]> on host b5c40a6f08c7
[2021-04-26 09:08:52,808] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-08-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-08-01T00:00:00+00:00
[2021-04-26 09:08:52,809] {stage_redshift.py:63} INFO - Start implementation of StageToRedshiftOperator
[2021-04-26 09:08:52,810] {stage_redshift.py:68} INFO - Preparing destination Redshift table
[2021-04-26 09:08:53,206] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 09:08:54,723] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-26 09:08:55,158] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-26 09:08:55,475] {stage_redshift.py:71} INFO - Getting data from S3 to Redshift
[2021-04-26 09:08:55,819] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 09:08:57,286] {dbapi.py:180} INFO - Running statement: 
             COPY staging_songs 
             FROM 's3://udacity-dend/song_data/*/*/*/'
             FORMAT AS JSON 'JSON 'auto''
             CREDENTIALS
             'aws_access_key_id=None;aws_secret_access_key=None'
                , parameters: None
[2021-04-26 09:08:57,688] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 81, in execute
    redshift.run(execution_sql)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/hooks/dbapi.py", line 184, in run
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "auto"
LINE 4:              FORMAT AS JSON 'JSON 'auto''
                                           ^

[2021-04-26 09:08:57,699] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200801T000000, start_date=20210426T090848, end_date=20210426T090857
[2021-04-26 09:08:58,374] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-26 12:36:02,299] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-26 12:36:02,655] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-26 12:36:02,657] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 12:36:02,658] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-26 12:36:02,658] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-26 12:36:03,001] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-08-01T00:00:00+00:00
[2021-04-26 12:36:03,526] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-08-01T00:00:00+00:00', '--job-id', '22', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmp2rl4ug94', '--error-file', '/tmp/tmpqucgozvv']
[2021-04-26 12:36:03,539] {standard_task_runner.py:77} INFO - Job 22: Subtask Stage_songs
[2021-04-26 12:36:03,201] {standard_task_runner.py:52} INFO - Started process 104 to run task
[2021-04-26 12:36:06,911] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [running]> on host 0f1e1da1f4f6
[2021-04-26 12:36:10,903] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-08-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-08-01T00:00:00+00:00
[2021-04-26 12:36:11,231] {stage_redshift.py:62} INFO - Start implementation of StageToRedshiftOperator
[2021-04-26 12:36:11,233] {stage_redshift.py:67} INFO - Preparing destination Redshift table
[2021-04-26 12:36:11,384] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 12:36:12,881] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-26 12:36:13,322] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-26 12:36:13,618] {stage_redshift.py:70} INFO - Getting data from S3 to Redshift
[2021-04-26 12:36:13,859] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-26 12:36:15,867] {dbapi.py:180} INFO - Running statement: 
             COPY staging_songs 
             FROM '/offor_data/song_data/*/*/*/'
             FORMAT AS JSON 'JSON 'auto''
                  , parameters: None
[2021-04-26 12:36:19,054] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 80, in execute
    redshift.run(execution_sql)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/hooks/dbapi.py", line 184, in run
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "auto"
LINE 4:              FORMAT AS JSON 'JSON 'auto''
                                           ^

[2021-04-26 12:36:19,666] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200801T000000, start_date=20210426T123602, end_date=20210426T123619
[2021-04-26 12:36:20,318] {local_task_job.py:146} INFO - Task exited with return code 1
[2021-04-27 20:39:53,581] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-27 20:39:54,244] {taskinstance.py:877} INFO - Dependencies all met for <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [queued]>
[2021-04-27 20:39:54,244] {taskinstance.py:1068} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 20:39:54,245] {taskinstance.py:1069} INFO - Starting attempt 1 of 2
[2021-04-27 20:39:54,245] {taskinstance.py:1070} INFO - 
--------------------------------------------------------------------------------
[2021-04-27 20:39:54,770] {taskinstance.py:1089} INFO - Executing <Task(StageToRedshiftOperator): Stage_songs> on 2020-08-01T00:00:00+00:00
[2021-04-27 20:39:54,963] {standard_task_runner.py:52} INFO - Started process 223 to run task
[2021-04-27 20:39:55,211] {standard_task_runner.py:76} INFO - Running: ['airflow', 'tasks', 'run', 'project_dag', 'Stage_songs', '2020-08-01T00:00:00+00:00', '--job-id', '92', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/udac_example_dag.py', '--cfg-path', '/tmp/tmp9tdjegvn', '--error-file', '/tmp/tmpkw0p1v2w']
[2021-04-27 20:39:55,272] {standard_task_runner.py:77} INFO - Job 92: Subtask Stage_songs
[2021-04-27 20:39:57,832] {logging_mixin.py:104} INFO - Running <TaskInstance: project_dag.Stage_songs 2020-08-01T00:00:00+00:00 [running]> on host 478c7ec349b0
[2021-04-27 20:40:00,044] {taskinstance.py:1283} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=ernestoffor@gmail.com.com
AIRFLOW_CTX_DAG_OWNER=udacity
AIRFLOW_CTX_DAG_ID=project_dag
AIRFLOW_CTX_TASK_ID=Stage_songs
AIRFLOW_CTX_EXECUTION_DATE=2020-08-01T00:00:00+00:00
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2020-08-01T00:00:00+00:00
[2021-04-27 20:40:00,045] {stage_redshift.py:67} INFO - Start implementation of StageToRedshiftOperator
[2021-04-27 20:40:00,046] {stage_redshift.py:74} INFO - Preparing destination Redshift table
[2021-04-27 20:40:00,444] {base.py:78} INFO - Using connection to: id: redshift. Host: redshift.cxvwoeu3kuea.us-west-2.redshift.amazonaws.com, Port: 5439, Schema: dev, Login: awsuser, Password: XXXXXXXX, extra: None
[2021-04-27 20:40:01,898] {dbapi.py:180} INFO - Running statement: DELETE FROM staging_songs, parameters: None
[2021-04-27 20:40:02,335] {dbapi.py:186} INFO - Rows affected: 0
[2021-04-27 20:40:02,645] {stage_redshift.py:77} INFO - Getting data from S3 to Redshift
[2021-04-27 20:40:05,181] {taskinstance.py:1482} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1138, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1311, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/models/taskinstance.py", line 1341, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/plugins/operators/stage_redshift.py", line 83, in execute
    s3.get_credentials().access_key,
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/amazon/aws/hooks/base_aws.py", line 478, in get_credentials
    return session.get_credentials().get_frozen_credentials()
AttributeError: 'NoneType' object has no attribute 'get_frozen_credentials'
[2021-04-27 20:40:05,197] {taskinstance.py:1532} INFO - Marking task as UP_FOR_RETRY. dag_id=project_dag, task_id=Stage_songs, execution_date=20200801T000000, start_date=20210427T203953, end_date=20210427T204005
[2021-04-27 20:40:06,123] {local_task_job.py:146} INFO - Task exited with return code 1
